### ISSUE TITLE
WITH... SELECT not behaving as expected

### ISSUE THREAD

**Issue by rzhang86** [ClickHouse/ClickHouse#16678]
Query:
WITH t1 AS (SELECT 1) SELECT 2

Expected result:
2

Actual result:
Syntax error: failed at position 12 ('('):
WITH t1 AS (SELECT 1) SELECT 2
Expected identifier

Query was constructed as per Example 4 from online docs (https://clickhouse.tech/docs/en/sql-reference/statements/select/with/). Why doesn't it work?

EDIT: Note, this example uses a scalar subquery, but as per Example 4 in the docs this should also work for subqueries returning more than one row. i.e., this should also work but does not:
WITH t1 AS (SELECT * FROM table1) SELECT * FROM t1

**Comment by den-crane** [comment]
This feature will be released in 20.11.
Documentation reflects the master code branch.

```

ClickHouse client version 20.11.1.5065.
Connecting to localhost:9000 as user default.
Connected to ClickHouse server version 20.11.1 revision 54442.

 WITH t1 AS (SELECT 1) SELECT 2

WITH t1 AS
    (
        SELECT 1
    )
SELECT 2

Query id: 5b445ecb-cf93-4b65-b6eb-4c84fb523b3b

┌─2─┐
│ 2 │
└───┘

1 rows in set. Elapsed: 0.001 sec.

```

**Comment by den-crane** [comment]
I am wrong.

It is supported in the stable already.

```sql
ClickHouse client version 20.10.3.30 (official build).
Connecting to localhost:9000 as user default.
Connected to ClickHouse server version 20.10.3 revision 54441.

8a861f168fda :)  WITH t1 AS (SELECT 1) SELECT 2

WITH t1 AS
    (
        SELECT 1
    )
SELECT 2

┌─2─┐
│ 2 │
└───┘
```
